File: 안유진/10828.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args)throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        Stack<Integer> stack = new Stack<>();

        int N = Integer.parseInt(br.readLine());

        for(int i = 0; i<N; i++){
            String s = br.readLine();
            StringTokenizer st = new StringTokenizer(s," ");

            String comm = st.nextToken();

            if(st.hasMoreTokens()){
                int num = Integer.parseInt(st.nextToken());
                stack.push(num);
            }

            if(comm.equals("pop")){
                if(stack.empty()){
                    System.out.println(-1);
                }else{
                    System.out.println(stack.pop());
                }
            }

            if(comm.equals("size")){
                System.out.println(stack.size());
            }

            if(comm.equals("empty")){
                if(stack.empty()){
                    System.out.println(1);
                }else {
                    System.out.println(0);
                }
            }

            if(comm.equals("top")){
                if(stack.empty()){
                    System.out.println(-1);
                }else{
                    System.out.println(stack.lastElement());
                }
            }

        }
    }
}

File: 송영운/temp/10828.py
class stack:
    def __init__(self):
        self.s = list()
        self.len = 0
    def push(self, num):
        self.len += 1
        self.s.append(num)

    def pop(self):
        if self.len > 0:
            self.len -= 1
            return self.s.pop()
        else:
            return -1
    def top(self):
        if self.len > 0:
            return self.s[-1]
        else:
            return -1
    def size(self):
        return self.len
    def empty(self):
        if self.len == 0:
            return 1
        else:
            return 0

input = __import__('sys').stdin.readline
if __name__ == '__main__':
    n = int(input())
    s = stack()
    for _ in range(n):
        line = list(input().split())
        if len(line) == 1:
            instruction = line[0]
        else:
            instruction, num = line[0], line[1]
        if instruction == 'push':
            s.push(num)
        elif instruction == 'pop':
            print(s.pop())
        elif instruction == 'top':
            print(s.top())
        elif instruction == 'size':
            print(s.size())
        elif instruction == 'empty':
            print(s.empty())


